### PR TITLE
2.1 updates

### LINKED ISSUE #146: [PE-D][Tester B] Stalemate
**Note from the teaching team:** This bug was reported during the _Part II (Evaluating Documents)_ stage of the PE. **You may reject this bug if it is not related to the quality of documentation.**<hr>
Inability to test feature of what happens during a stalemate as player is unable to get to that scenario.

-------------
Labels: `severity.Medium` `type.FeatureFlaw`
original: TeoHaoZhi/ped#8

### PR DESCRIPTION
**Pull request by onx001**

### CODE REVIEW

File: src/main/java/chessmaster/game/ChessBoard.java [hunk 1]
```diff
@@ -395,8 +396,9 @@ public boolean isEndGame() {
     }
 
     public Color getWinningColor() {
-        
-        if (isCheckmated(playerColor)) {
+        if (isCheckmated(playerColor) && isCheckmated(playerColor.getOppositeColour())) {
+            return Color.DRAW;
```
Comment (antrikshdhand) [inline src/main/java/chessmaster/game/ChessBoard.java:400]: So it only draws the game if _both_ players are checkmated at the same time? Will this condition ever be true? To be honest I'm not entirely sure how you would define stalemate – I thought it was when one player isn't in checkmate but can't make any moves.

**Review by antrikshdhand — COMMENTED** [review]
`mostPoints()` is definitely more readable now, though making a hard algorithm like that easy to understand will always be difficult. Have you tested the draw condition?

File: src/main/java/chessmaster/game/ChessBoard.java (hunk 1 above)
Comment (onx001) [reply src/main/java/chessmaster/game/ChessBoard.java:400]: Here, the isCheckmated function searches for any available moves. A stalemate can also occur when there are no available moves for either party, even though neither might be technically in check.

Comment (antrikshdhand) [reply src/main/java/chessmaster/game/ChessBoard.java:400]: Sounds good.

**Review by antrikshdhand — APPROVED** [review]

**Pull request merged**